torch/_inductor/__init__.py

```diff
@@ -96,9 +96,12 @@ def aoti_compile_and_package(
     assert isinstance(m, torch.fx.GraphModule)
 
     aoti_files = aot_compile(m, args, kwargs, options=inductor_configs)  # type: ignore[arg-type]
+    assert isinstance(aoti_files, list)
+
+    so_path = next(file for file in aoti_files if file.endswith((".so", ".cpp")))
 
     if package_path is None:
-        package_path = aoti_files + ".pt2"
+        package_path = f"{so_path[:-3]}.pt2"
```

Comment:
Either use Pathlib tools or rsplit, which is much safer that hardcoding the suffix size (unless you want to assert beforehand that `so_path.ends_with(".so")`)
```suggestion
        package_path = f"{so_path.rsplit('.', 1)[0]}.pt2"
```